Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java

```diff
@@ -1706,6 +1711,66 @@ public void run() {
         }
     }
 
+    @JavascriptInterface
+    public void initialWebViewSetUp(String callback, String id) {
+        webViewCallBack = callback;
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                WebView webView= (WebView) activity.findViewById(Integer.parseInt(id));
+                webView.setWebViewClient(new WebViewClient(){
+                    @Override
+                    public boolean shouldOverrideUrlLoading(WebView view, String url) {
```

Comment:
@kapilJhalani1 why this is required ? As we can see it is deprecated from API 24 which means it will not support the major versions which our app supports.

Reply:
Android Webview can only handle http and https url's, so for handling other url's  we need to separately handle them.

Reply:
Is this a deprecated function @kapilJhalani1 ??

Reply:
Yes in API level above 24 ' shouldOverrideUrlLoading(WebView view, WebResourceRequest request)' this function is getting used but it is not compatible with the older versions.

And the older function ' shouldOverrideUrlLoading(WebView view, String url) ' is also good with new versions as backward compatibility is there.